.github/repo_analysis.ipynb

```diff
@@ -0,0 +1,194 @@
+{
```

Comment:
Btw. something seems to be wrong with the rebase, at least other files are now shown in GitHub. Other than that, I am basically happy with any of the proposed syntax, important is that it helps you best for improving test automation.

Reply:
I will look over this PR again tomorrow.

Reply:
The best for improving test automation is to write the feature files directly into e2e/features, assuming it is automatable, and developers reuse already implemented steps.
Everything else doesnt have that much of an impact on automation (for me at least), as I would still need to go write and probably correct some of the syntax of that feature file. In that case I would rather prefer readability in the tests protocol.
But if writing test protocols benefits others in learning the syntax and keeping the protocol short, thats maybe something worth to consider.

Reply:
So I have mirrored each doc/usecases/done markdown into one feature file. Additionally we have some feature files that should prevent regressions (base_layer, plant_selection).
Maybe we can create the directories e2e/features/usecases and e2e/features/regressions

Maybe we could also place the features inside doc/features/usecases and refactor all markdowns to gherkin syntax.

Reply:
> Maybe we can create the directories e2e/features/usecases and e2e/features/regressions

And in the usecases folder a folder or file per usecase?

> Maybe we could also place the features inside doc/features/usecases and

It could be in doc but maybe `doc/scenarios` or maybe even directly within the use cases?

> refactor all markdowns to gherkin syntax.

I wouldn't do that now, as it is:

1. on a different level with different focus (use cases vs. scenarios)
2. a lot of effort

I think they better should be side-by-side for now, at least until we haven't implemented all the use cases (the feature files are not complete).

> Also it is possible to write scenarios that are untested yet, so we could define error cases and non-functional constraints without implementing them (yet).

Yes, very good idea! :heart:

> And also have doc/features/regressions ?

I am not sure if the regression/use case distinction needs to be on folder level. Also regressions can be assigned to use cases, can't they?